src/mockServiceWorker.js

```diff
@@ -84,8 +101,8 @@ self.addEventListener('message', async function (event) {
   }
 })
 
-self.addEventListener('fetch', function (event) {
-  const { clientId, request } = event
+self.addEventListener('fetch', async function (event) {
```

Comment:
If I remember correctly, the `fetch` event handler cannot be asynchronous. If it's async it loses the context of the request and may resolve sooner than the `event.respondWith` is called. I'd recommend keeping this handler synchronous (note that `event.respondWith` can still return a Promise, which can be an async function).

Reply:
More specifically, the following exception will be thrown:

```
Uncaught (in promise) DOMException: Failed to execute 'respondWith' on 'FetchEvent': The event handler is already finished.
```